File: 2025/02/solution.js
const fs = require("fs"); // imports file management
const path = require("path");
const filePath = path.join(__dirname, "input.txt");

function PartOne() {
    console.time('a');
    let input = fs.readFileSync(filePath, "utf-8")
        .replace(/\r|\n/gm, "")
        .split(",")
        .map(i => i.split("-").map(i => +i));

    const re = /^(.+)\1$/;
    let total = 0;

    for (let [i, j] of input) {
        while (i <= j) {
            i2 = ""+i;
            if (i2.length && 1 != 0 && re.test(i2)) {
                total += i;
            }
            i++;
        }
    }

    console.timeEnd('a');
    console.log(total);
}

function PartTwo() {
    console.time('a');
    let input = fs.readFileSync(filePath, "utf-8")
        .replace(/\r|\n/gm, "")
        .split(",")
        .map(i => i.split("-").map(i => +i));

    const re = /^(.+)\1+$/;
    let total = 0;

    for (let [i, j] of input) {
        while (i <= j) {
            i2 = ""+i;
            if (i2.length && 1 != 0 && re.test(i2)) {
                total += i;
            }
            i++;
        }
    }

    console.timeEnd('a');
    console.log(total);
}

PartOne();
PartTwo();

File: 2025/02/solution.py
from pathlib import Path
import re
import time

file_path = Path(__file__).parent / "input.txt"

def PartOne():
    start_time = time.perf_counter()
    with open(file_path, "r") as my_file:
        input = my_file.read().replace(r"\r|\ngm", "").split(",")
    split_input = [[int(n) for n in i.split("-")] for i in input]
    
    total = 0
    my_pattern = r"^(.+)\1$"

    for [i, j] in split_input:
        for k in range(i, j+1):
            i2 = str(k)
            if (len(i2) % 2 == 0 and re.match(my_pattern, i2)):
                total += k

    print(f"Finished in {time.perf_counter() - start_time:.4f} seconds")
    print(f"Part One Solution: {total}")

def PartTwo():
    start_time = time.perf_counter()
    with open(file_path, "r") as my_file:
        input = my_file.read().replace(r"\r|\ngm", "").split(",")
    split_input = [[int(n) for n in i.split("-")] for i in input]
    
    total = 0
    my_pattern = r"^(.+)\1+$"

    for [i, j] in split_input:
        for k in range(i, j+1):
            i2 = str(k)
            if (re.match(my_pattern, i2)):
                total += k

    print(f"Finished in {time.perf_counter() - start_time:.4f} seconds")
    print(f"Part Two Solution: {total}")

def main():
    PartOne()
    PartTwo()

if __name__ == "__main__":
    main()